applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -486,21 +478,16 @@ static void disconnected_cb(struct bt_conn *conn, uint8_t reason)
 	uint8_t conn_index;
 	char addr[BT_ADDR_LE_STR_LEN];
 
-	if (conn != headset_conn[AUDIO_CHANNEL_LEFT] && conn != headset_conn[AUDIO_CHANNEL_RIGHT]) {
-		LOG_ERR("Unknown connection disconnected, should not reach here");
-		return;
-	}
-
 	(void)bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
 
 	LOG_INF("Disconnected: %s (reason 0x%02x)", addr, reason);
 
-	ret = headset_conn_index_get(conn, &conn_index);
+	bt_conn_unref(conn);
 
+	ret = headset_conn_index_get(conn, &conn_index);
 	if (ret) {
-		LOG_ERR("Unknown connection, should not reach here");
+		LOG_WRN("Unknown connection, should not reach here");
```

Comment:
```suggestion
		LOG_WRN("Unknown connection");
```
Since this is a possibility now